src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -0,0 +1,150 @@
+            GetBooleanDetails(flagKey, defaultValue, context, config).Value;
+
+        public FlagEvaluationDetails<bool> GetBooleanDetails(string flagKey, bool defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null) =>
+            EvaluateFlag(flagKey, defaultValue, FlagValueType.Boolean, context, config);
+
+        public string GetStringValue(string flagKey, string defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null) =>
+            GetStringDetails(flagKey, defaultValue, context, config).Value;
+        
+        public FlagEvaluationDetails<string> GetStringDetails(string flagKey, string defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null) =>
+            EvaluateFlag(flagKey, defaultValue, FlagValueType.String, context, config);
+
+        public int GetNumberValue(string flagKey, int defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null) =>
+            GetNumberDetails(flagKey, defaultValue, context, config).Value;
+        
+        public FlagEvaluationDetails<int> GetNumberDetails(string flagKey, int defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null) =>
+            EvaluateFlag(flagKey, defaultValue, FlagValueType.Number, context, config);
+
+        public T GetObjectValue<T>(string flagKey, T defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null)  =>
+            GetObjectDetails(flagKey, defaultValue, context, config).Value;
+
+        public FlagEvaluationDetails<T> GetObjectDetails<T>(string flagKey, T defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null)  =>
+            EvaluateFlag(flagKey, defaultValue, FlagValueType.Object, context, config);
+
+
+        private FlagEvaluationDetails<T> EvaluateFlag<T>(string flagKey, T defaultValue, FlagValueType flagValueType, EvaluationContext context = null, FlagEvaluationOptions options = null)
+        {
+            var provider = OpenFeature.GetProvider();
+         
+            // New up a evaluation context if one was not provided.
+            if (context == null) 
+            {
+                context = new EvaluationContext();
+            }
+
+            /*var allHooks = new List<IHook>()
+                .Concat(OpenFeature.GetHooks())
+                .Concat(_hooks)
+                .ToList();
+            
+            var allHooksReversed = allHooks
+                .AsEnumerable()
+                .Reverse()
+                .ToList();
+            
+            var hookContext = new HookContext<bool>(
+                flagKey,
+                defaultValue,
+                FlagValueType.Boolean,
+                _metadata,
+                OpenFeature.GetProviderMetadata(),
+                context
+            );*/

```

Comment:
Looks like this is commented on purpose, b/c you haven't implemented hooks, yet, but just commenting to make sure you didn't mean to remove it.

Reply:
Yea i have got through implementing hooks and decided I should just get the basics done first. Ill rebase and remove this out for now